### PR TITLE
Extend test coverage and add NatSpec comments

### PR DESCRIPTION
**Pull request by Bielwenass**
Implemented the missing tests, including the upgradeable contract one, and added NatSpec annotations. Please let me know if any changes are needed.

### CODE REVIEW

File: src/contracts/IdleRouter.sol [hunk 1]
```diff
@@ -56,18 +64,40 @@ contract IdleRouter is OwnableUpgradeable {
         _deposit(_token, _amount, true);
     }
 
+    /**
+     * @notice deposit tokens into the AA CDO Tranche. Note: approvals must be made before
```
Comment (0xean) [inline src/contracts/IdleRouter.sol]: This one should be the BB CDO Tranche

File: src/contracts/IdleRouter.sol [hunk 2]
```diff
@@ -56,18 +64,40 @@ contract IdleRouter is OwnableUpgradeable {
...
+     * this call to allow the router to move your assets on your behalf
+     * @param _token the underlying token for the CDO
+     * @param _amount of the underlying token to deposit
+     */
     function depositBB(address _token, uint256 _amount) external {
         _deposit(_token, _amount, false);
     }
 
+    /**
+     * @notice burn AA Tranche tokens and get the principal + interest back
```
Comment (0xean) [inline src/contracts/IdleRouter.sol]: Might want to add a note here also that users will need to grant the router approval to move its _tranceToken as well.

File: test/contracts/idleRouterTests.js [hunk 3]
```diff
@@ -2,11 +2,11 @@ const { expect } = require("chai");
 const { ethers, upgrades } = require("hardhat");
 const ERC20 = require("@openzeppelin/contracts/build/contracts/ERC20.json");
 
-DAI_CDO_ADDRESS = "0xd0DbcD556cA22d3f3c142e9a3220053FD7a247BC";
-DAI_CDO_AA_TRANCHE = "0xe9ada97bdb86d827ecbaacca63ebcd8201d8b12e";
-DAI_CDO_BB_TRANCHE = "0x730348a54ba58f64295154f0662a08cbde1225c2";
-DAI_ADDRESS = "0x6B175474E89094C44Da98b954EedeAC495271d0F";
-DAI_WHALE = "0xe78388b4ce79068e89bf8aa7f218ef6b9ab0e9d0";
+const DAI_CDO_ADDRESS = "0xd0DbcD556cA22d3f3c142e9a3220053FD7a247BC";
```
Comment (0xean) [inline test/contracts/idleRouterTests.js:5]: TY or adding this! hah my bad.

File: test/contracts/idleRouterTests.js [hunk 4]
```diff
@@ -318,15 +419,75 @@ describe("IdleRouter", () => {
...
-    // upgrade our existing implementation and check that basic functionality works.
-    // the V2 can be live in the mocks folder and simply ad a new state variable or something.
-    // this can be done with HH upgrades with the following example.
-    // const IdleRouterV2 = await ethers.getContractFactory("IdleRouterV2");
-    // idleRouterV2 = await upgrades.upgradeProxy(
-    //   idleRouter.address,
-    //   IdleRouterV2
-    // );
-    // await idleRouterV2.deployed();
+    it("upgrades successfully", async () => {
```
Comment (0xean) [inline test/contracts/idleRouterTests.js:422]: 👏  - very nice!

**Review by 0xean — APPROVED** [review]
Some minor nitpicks on the comments but this looks great!

**Comment by Bielwenass** [comment]
> Some minor nitpicks on the comments but this looks great!

Appreciate the feedback! Pushed the comment fixes

**Review by 0xean — APPROVED** [review]

**Comment by 0xean** [comment]
Thanks @Bielwenass!

Closes #1

**Pull request merged**